hibernate-core/src/main/java/org/hibernate/boot/registry/internal/BootstrapServiceRegistryImpl.java

```diff
@@ -235,23 +235,22 @@ public ServiceRegistry getParentServiceRegistry() {
 	}
 
 	@Override
-	public <R extends Service> void configureService(ServiceBinding<R> binding) {
+	public <R extends Service> void configureService(R service) {
 		throw new ServiceException( "Boot-strap registry should only contain provided services" );
 	}
 
 	@Override
-	public <R extends Service> void injectDependencies(ServiceBinding<R> binding) {
+	public <R extends Service> void injectDependencies(R service) {
 		throw new ServiceException( "Boot-strap registry should only contain provided services" );
 	}
 
 	@Override
-	public <R extends Service> void startService(ServiceBinding<R> binding) {
+	public <R extends Service> void startService(R service, Class<R> serviceRole) {
```

Comment:
I'm not sure that we need to change the signature here. All these methods are called from a synchronized block, so we could do something like this:

```
final R tempService = lifecycleOwner.initiateService( serviceInitiator );

service = tempService;

if ( tempService != null ) {
    lifecycleOwner.injectDependencies( this );
    lifecycleOwner.configureService( this );
    lifecycleOwner.startService( this );
    lifecycleOwner.registerService( this );
}
```

This way, we can pass the `ServiceBinding` and each method can access the currently built Service along with some other info (e.g. serviceRole). The signature is more consistent this way, and we can better accommodate other objects that might need to be passed to these method in future.